### ISSUE TITLE
[unbound-method]: should ignore accessing `jest.mocked().mock`

### ISSUE THREAD

**Issue by SchroederSteffen** [jest-community/eslint-plugin-jest#1800]
I noticed that the `unbound-method` rule reports the following code:
```ts
const parameter = jest.mocked(service.method).mock.calls[0][0];
```

However, the following variant is not reported:
```ts
const mockedMethod = jest.mocked(service.method);
const parameter = mockedMethod.mock.calls[0][0];
```

Using any `.mockXY()` methods is also fine (as expected since #1596):
```ts
jest.mocked(service.method).mockReturnValue();
```

<img width="1778" height="897" alt="Image" src="https://github.com/user-attachments/assets/cbeb4c49-ad8c-40b4-9bf6-32330aa58caf" />

Reproduction: https://github.com/SchroederSteffen/jest-eslint-unbound-method-repro/blob/main/src/service.spec.ts

I think the mentioned case should NOT report an error.

**Comment by G-Rath** [comment]
yeah I think this is reasonable
